src/vm/appdomainnative.cpp

```diff
@@ -253,26 +139,6 @@ FCIMPL0(Object*, AppDomainNative::GetLoadedAssemblies)
 } // AppDomainNative::GetAssemblies
 FCIMPLEND
 
-FCIMPL1(INT32, AppDomainNative::GetId, AppDomainBaseObject* refThisUNSAFE)
-{
-    FCALL_CONTRACT;
-
-    INT32        iRetVal = 0;
-    APPDOMAINREF refThis = (APPDOMAINREF) refThisUNSAFE;
-
-    HELPER_METHOD_FRAME_BEGIN_RET_1(refThis);
-
-    AppDomain* pApp = ValidateArg(refThis);
-    // can only be accessed from within current domain
-    _ASSERTE(GetThread()->GetDomain() == pApp);
-
-    iRetVal = pApp->GetId().m_dwId;
```

Comment:
We could also remove all the logic around assigning AppDomain ids, the m_dwId field and so on - but that in itself looks like a good sized change on its own.

Reply:
Yes, a lot of opportunities for cleanup. I have to always remind myself to stop going to get a reasonable sized stable change.